web_app/models/AC_model_iso9223.py

```diff
@@ -112,29 +113,39 @@ def _determine_exponent(self) -> float:
 
     def eval_material_loss(self, time: float) -> float:
         """Calculates and returns the material loss over time based on the provided parameters."""
-
         # Calculate corrosion speed if not provided
         if 'corrosion_speed' in self.parameters:
-            corrosion_speed = self.parameters['corrosion_speed']
+            corrosion_speed = self.g_to_um_map(self.parameters['corrosion_speed'])
+            material_loss = time*corrosion_speed
         else:
             if self.parameters['T'] <= 10:
                 fst = 0.15 * (self.parameters['T'] - 10)
             else:
                 fst = -0.054 * (self.parameters['T'] - 10)
-
             corrosion_speed = (1.77 * self.parameters['Pd'] ** 0.52 * np.exp(0.02 * self.parameters['RH'] + fst) +
-                               0.102 * self.parameters['Sd'] ** 0.62 * np.exp(
-                                   0.033 * self.parameters['RH'] + 0.04 * self.parameters['T']))
-
-        # Calculate material loss over time
-        if np.max(time) < 20:
+                   0.102 * self.parameters['Sd'] ** 0.62 * np.exp(
+                       0.033 * self.parameters['RH'] + 0.04 * self.parameters['T']))
+            corrosion_speed /= 1000  #Convert to grams from milli grams
+            corrosion_speed = self.g_to_um_map(corrosion_speed)
             material_loss = self.parameters['exponent'] * corrosion_speed * time ** (self.parameters['exponent'] - 1)
-        else:
-            material_loss = corrosion_speed * (20 ** self.parameters['exponent'] +
-                                               self.parameters['exponent'] * 20 ** (self.parameters['exponent'] - 1) * (
-                                                           time - 20))
 
+        st.write('Corrosion speed = ' + str(corrosion_speed))
+        
         return material_loss
```

Comment:
unclear naming between corrosion_speed in g and um

Reply:
changed the name to grams_to_um_map